### PR TITLE
Add GQL API for social volume

### PR DESCRIPTION
**Pull request by LucoLe**
#### Summary

Adds a GQL endpoint for social volume.
Updates the apidocs to document the endpoint use.
Adds tests for the new functions in internal_services/tech_indicators.ex

[//]: # (#### Related PRs)

[//]: # (#### Additional deploy notes)

#### Screenshots

image

image

image

### CODE REVIEW

File: test/sanbase/internal_services/tech_indicators_test.exs [hunk 1]
```diff
@@ -226,4 +226,101 @@ defmodule Sanbase.InternalServices.TechIndicatorsTest do
...
+                    datetime: DateTime.from_unix!(to)
+                  }
+                ]}
+    end
+
+    test "response: 404" do
+      from = 1_523_876_400
+      to = 1_523_880_000
+
+      mock(
```
Comment (LucoLe) [inline test/sanbase/internal_services/tech_indicators_test.exs]: Unfortunately, this generates an error that is logged when the tests are run. Do you think that is too messy or it's ok @valo @IvanIvanoff.

Comment (IvanIvanoff) [reply test/sanbase/internal_services/tech_indicators_test.exs]: Yes, it's an issue to leave error logs in the tests. For this case you should use `capture_log` from the `ExUnit.CaptureLog` module

File: test/sanbase/internal_services/tech_indicators_test.exs [hunk 2]
```diff
@@ -226,4 +226,101 @@ defmodule Sanbase.InternalServices.TechIndicatorsTest do
...
+      result =
+        TechIndicators.social_volume(
+          "Foo",
+          DateTime.from_unix!(from),
+          DateTime.from_unix!(to),
+          "1h",
+          :telegram_discussion_overview
+        )
+
+      assert result == {:error, "[#{log_id}] Error executing query. See logs for details."}
```
Comment (IvanIvanoff) [inline test/sanbase/internal_services/tech_indicators_test.exs]: Mocking the `Ecto.UUID` complicates the code without any benefits. For here you should not mock and instead:
```elixir
assert {:error, err_msg} = result
assert err_msg =~ "Error executing query. See logs for details."
```

**Comment by valo** [comment]
Now that I am looking into this I realize that we should be fetching these social volumes by slug, not by ticker. We are trying to deprecate the tickers as identifiers and use the slugs. This will make integration of the APIs in the google spreadsheet easier too.

**Comment by LucoLe** [comment]
yeah, I was afraid you'd say that :) but this will require a change in tech indicators and I'd rather do that in a separate PR/task. Let's finalise this PR and I'll create a new task in Favro and take right after this one.

**Comment by valo** [comment]
I think it will be better if we make this API public only after we use slugs. Otherwise we are releasing an API and then immediately introduce a breaking change. May be remove the official docs for now and add them later?

**Comment by LucoLe** [comment]
ok let me see how much work it will be to make the changes in tech indicators

File: lib/sanbase_web/graphql/schema.ex [hunk 3]
```diff
@@ -431,7 +431,7 @@ defmodule SanbaseWeb.Graphql.Schema do
         It is used to select the source of the mentions count.
     """
     field :social_volume, list_of(:social_volume) do
-      arg(:ticker, non_null(:string))
+      arg(:project, non_null(:string))
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/schema.ex]: let's not use `project` as it's reserved for the `project` GQL type. Something like `:ticker_slug` should be better here.

Comment (IvanIvanoff) [reply lib/sanbase_web/graphql/schema.ex]: The price measurement name is `SAN_santiment` not just `santiment`, hence the `ticker` in the `ticker_slug`

**Comment by valo** [comment]
Why do we need to use `SAN_santiment` instead of `santiment`?

**Comment by IvanIvanoff** [comment]
@valo when deciding how to name the new price measurements we aggreed on `ticker_slug` as many people recognize the projects mainly by tickers and by their coinmarketcap ids.

The main question is why not use just `slug` here. I guess it's momentum аfter the work on tech_indicators.
@LucoLe it's only needed to query `slug` here. The `ticker` we can get from the DB - check how it's made in all other queries.

**Comment by valo** [comment]
Do we use ticker_slug construct for other metrics? I think for the others we use only slug. That’s why I am asking how is this one different

**Comment by LucoLe** [comment]
We have a dictionary in tech_indicators that keeps the search strings. The keys in this dictionary need to be in the form `ticker_slug` because we use this key in Grafana to make a query for the price in influxDB we use either `ticker` or `ticker_slug` we can't use only `slug`. So this is the reason why we have the projects in this format. If you think that's too cumbersome we have only one option I can see to serve the prices from tech indicators to Grafana. This way we can change the keys to whatever we want.

**Comment by IvanIvanoff** [comment]
@LucoLe Does grafana/tech_indicators make GQL query to this endpoint? If no - then better follow the convention here and not mimic the grafana/tech_indicators behaviour (which is such because it's a workaround)

File: lib/sanbase/internal_services/tech_indicators.ex [hunk 4]
```diff
@@ -447,6 +447,10 @@ defmodule Sanbase.InternalServices.TechIndicators do
     from_unix = DateTime.to_unix(datetime_from)
     to_unix = DateTime.to_unix(datetime_to)
 
+    ticker_slug =
+      Repo.get_by(Project, coinmarketcap_id: slug)
+      |> Measurement.name_from()
```
Comment (IvanIvanoff) [inline lib/sanbase/internal_services/tech_indicators.ex]: In that case it's an overhead to fetch the whole project just to use the ticker. I suggest using `ticker_by_slug` for this case. I would rather:
```
ticker = Project.ticker_by_slug(slug)
ticker_cmc_id = "#{ticker}_#{slug}"

File: lib/sanbase_web/graphql/schema.ex [hunk 5]
```diff
@@ -416,6 +416,43 @@ defmodule SanbaseWeb.Graphql.Schema do
       resolve(&TechIndicatorsResolver.emojis_sentiment/3)
     end
 
+    @desc ~s"""
+    Returns a list of mentions count for a given project and time interval.
+
+    Arguments description:
+      * slug - a string
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/schema.ex]: `a string uniquely identifying a project`

File: lib/sanbase_web/graphql/schema.ex [hunk 6]
```diff
@@ -416,6 +416,43 @@ defmodule SanbaseWeb.Graphql.Schema do
...
+      * interval - an integer followed by one of: `m`, `h`, `d`
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/schema.ex]: also supported `w` for weeks

File: lib/sanbase_web/graphql/schema.ex [hunk 7]
```diff
@@ -416,6 +416,43 @@ defmodule SanbaseWeb.Graphql.Schema do
...
+      * from - a string representation of datetime value, e.g. "2018-04-16T10:02:19Z"
+      * to - a string representation of datetime value, e.g. "2018-04-16T10:02:19Z"
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/schema.ex]: We can just say that datetimes are following the iso8601 standard, so it's more clear what inputs we accept

**Pull request merged**